__init__.py

```diff
@@ -25,26 +26,53 @@ class RemarkablePlugin(DevicePlugin):
 
     MANAGES_DEVICE_PRESENCE = True
 
+    def __enter__(self, *args):
+        '''
+        Add this plugin to the python path so that it's contents become directly importable.
+        Useful when bundling large python libraries into the plugin. Use it like this::
+            with plugin:
+                import something
+        '''
+        if self.plugin_path is not None:
+            from calibre.utils.zipfile import ZipFile
+
+            with ZipFile(self.plugin_path) as zf:
+                # * Skipped check for zip_safe
+                from calibre.ptempfile import TemporaryDirectory
+                self._sys_insertion_tdir = TemporaryDirectory('plugin_unzip')
+                self.sys_insertion_path = self._sys_insertion_tdir.__enter__(*args)
+                zf.extractall(self.sys_insertion_path)
+                sys.path.append(self.sys_insertion_path)
```

Comment:
How is this fundamentally different from the [inherited implementation](https://github.com/kovidgoyal/calibre/blob/623b1d69f2a5d089a9a7f3a74e70f645d2f71d0c/src/calibre/customize/__init__.py#L280-L308), and why doesn't that work? The comment `skipped check for zip_safe` doesn't offer much to aid in one's understanding.